scanapp/views.py

```diff
@@ -151,3 +165,147 @@ def post(self, request):
                 }
             )
         )
+
+
+class URLFormView(FormView):
+    template_name = 'scanapp/urlspecialscan.html'
+    form_class = URLScanForm
+
+    def post(self, request, *args, **kwargs):
+        form = self.form_class(request.POST)
+
+        if form.is_valid():
+            # get the URL from the form
+            URL = request.POST['URL']
+
+            scan_type = 'URL'
+
+            allowed_exts = ('git')
+            url = urlparse(URL)
+
+            # Create an Instance of InsertIntoDB
+            insert_into_db = InsertIntoDB()
+
+            # call the create_scan_id function
+            scan_id = insert_into_db.create_scan_id(scan_type)
+
```

Comment:
Oh ! I'll add them in 2 different functions and also make sure that fetching is delegated to the background task.